server/app/services/program/ProgramService.java

```diff
@@ -1305,6 +1305,34 @@ public ProgramDefinition removeBlockPredicate(long programId, long blockDefiniti
     }
   }
 
+  /**
+   * Update the eligibility message for a block.
+   *
+   * @param programId the ID of the program to update
+   * @param blockDefinitionId the ID of the block to update
+   * @param message the custom eligibility message to add to the block
+   * @return the updated block
+   * @throws ProgramNotFoundException when programId does not correspond to a real Program.
+   * @throws ProgramBlockDefinitionNotFoundException when blockDefinitionId does not correspond to a
+   *     real Block.
+   */
+  public ProgramDefinition setBlockEligibilityMessage(
+      long programId, long blockDefinitionId, Optional<LocalizedStrings> message)
+      throws ProgramNotFoundException, ProgramBlockDefinitionNotFoundException {
+    ProgramDefinition programDefinition = getFullProgramDefinition(programId);
+
+    BlockDefinition blockDefinition =
+        programDefinition.getBlockDefinition(blockDefinitionId).toBuilder()
+            .setLocalizedEligibilityMessage(message)
+            .build();
+    try {
+      return updateProgramDefinitionWithBlockDefinition(programDefinition, blockDefinition);
+    } catch (IllegalPredicateOrderingException e) {
+      throw new RuntimeException(
+          "Unexpected error: setting this eligibility message invalidates another");
```

Comment:
```suggestion
       String errMsg = String.format("Setting this eligibility message invalidates another. [programId: %d, blockDefinitionId: %d]", programId, blockDefinitionId);
       throw new RuntimeException(errMsg, e);
```

(blocking) We should include the program and block ids and also add the caught exception to the new one.

Reply:
What does "invalidates another" mean?

Reply:
I read it as it invalidates another existing eligibility message, but on a second reading it would be nice to make the text more clear on what the issue is.

Reply:
Yup updated the error message, also added some comments to explain it more.